app/controllers/profiles_controller.rb

```diff
@@ -45,9 +45,25 @@ def confirm_phone
     end
   end
 
+  def change_password
+    @customer = current_customer
+  end
+
+  def update_password
+    @customer = current_customer
+    @customer.attributes = permitted_params
+    if @customer.update_password
+      redirect_to profile_path, notice: 'Password updated successfully'
+    else
+      respond_to do |format|
+        format.js {render partial: 'change_password_form', locals: {customer: @customer}, within: '#change-password-form-container'}
+      end
+    end
+  end
+
   private
   def permitted_params
```

Comment:
I think it's better to separate password params and params for updating other fields, if you use it in different forms.